dotnet/src/Experimental/Process.Core/ProcessStepBuilder.cs

```diff
@@ -194,13 +194,21 @@ protected ProcessStepBuilder(string name)
 /// </summary>
 public sealed class ProcessStepBuilder<TStep> : ProcessStepBuilder where TStep : KernelProcessStep
 {
+    /// <summary>
+    /// The initial state of the step. This may be null if the step does not have any state.
+    /// </summary>
+    private readonly object? _initialState;
+
     /// <summary>
     /// Creates a new instance of the <see cref="ProcessStepBuilder"/> class. If a name is not provided, the name will be derived from the type of the step.
     /// </summary>
-    public ProcessStepBuilder(string? name = null)
+    /// <param name="name">Optional: The name of the step.</param>
+    /// <param name="initialState">Optional: The initial state of the step.</param>
+    public ProcessStepBuilder(string? name = null, object? initialState = default)
```

Comment:
Observation - Its unfortunate that the type alignment isn't enforced at design-time; although, I like how the runtime failure has an actionable message.  Since I'd presume _state_ is only associated with the generic step-type  `KernelProcessStep<TState>`, I wonder if there's an opportunity to strengthen this pattern.  I recognize complicating the class heirarchy and using complex generic parameters is likely not be ideal.

Reply:
I also have been wondering about scoping...is the caller expected to instantiate their own `ProcessStepBuilder` classes or always go through the `ProcessBuilder`.  If the latter, perhaps `internal`?

Reply:
I agree, I tried to get it design time enforced but it explodes into generic parameters everywhere. We will take another look at this before we lock down the contracts.